core/profiling/build.cmd

```diff
@@ -0,0 +1,43 @@
+@echo off
+setlocal
+
+set VS_ENT_CMD_PATH="C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\VsDevCmd.bat"
+set VS_COM_CMD_PATH="C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\Tools\VsDevCmd.bat"
+
+if not defined VS_CMD_PATH (
+    if exist %VS_ENT_CMD_PATH% (
+        set VS_CMD_PATH=%VS_ENT_CMD_PATH%
+    ) else if exist %VS_COM_CMD_PATH% (
+        set VS_CMD_PATH=%VS_COM_CMD_PATH%
+    ) else (
+        echo No VS developer command prompt detected!
+        goto :EOF
+    )
+)
+
+echo.
+echo   Building
+
+if not exist bin\ (
+    mkdir bin
+)
+
+pushd bin
+
+cmake -G "Visual Studio 17 2022" ..\ -DCMAKE_BUILD_TYPE=Debug
+
+echo Calling VS Developer Command Prompt to build
+call %VS_CMD_PATH%
+
+msbuild -v:m ProfilerSamples.sln
```

Comment:
The real project file is CMakeLists.txt, using cmake allows it to build on any OS without having to manage a bunch of different project types. Since it is a C++ project if I checked in a sln file I would have to also check in a makefile for linux or have it only build for Windows.

Line 27 `cmake -G "Visual Studio 17 2022" ..\ -DCMAKE_BUILD_TYPE=Debug` uses cmake to generate a msbuild sln file, and in the corresponding build.sh there is a similar cmake command to generate a makefile.

The fact that it is a sln file is an implementation detail, you could also use ninja or any other build system cmake supports.